packages/plugin-dlx/sources/commands/dlx.ts

```diff
@@ -43,7 +74,35 @@ export default class DlxCommand extends BaseCommand {
 
       await xfs.writeFilePromise(ppath.join(tmpDir, toFilename(`package.json`)), `{}\n`);
       await xfs.writeFilePromise(ppath.join(tmpDir, toFilename(`yarn.lock`)), ``);
-      await xfs.writeFilePromise(ppath.join(tmpDir, toFilename(`.yarnrc.yml`)), `enableGlobalCache: true\n`);
+
+      const projectCwd = (await Configuration.findProjectCwd(this.context.cwd, Filename.lockfile))!;
+      const localConfigPath = ppath.join(projectCwd, toFilename(`.yarnrc.yml`));
+      const tmpDirConfigPath = ppath.join(tmpDir, toFilename(`.yarnrc.yml`));
+      if (await xfs.existsPromise(localConfigPath)) {
+        await xfs.copyFilePromise(localConfigPath, tmpDirConfigPath);
+        await Configuration.updateConfiguration(projectCwd, {
+          enableGlobalCache: false,
+        });
+
+        const bufferStream = new WritablePluginBufferStream();
+        await this.cli.run([`plugin`, `runtime`], {cwd: tmpDir, stdout: bufferStream});
+        const result = await bufferStream.finishPromise();
+
+        const {plugins, modules} = this.context.plugins;
+        const pluginNameArray = Array.from(plugins);
+
+        const allPluginConfigurationKeys = pluginNameArray.reduce<Array<string>>((allKeys, name) =>
+          [...allKeys, ...getConfigurationKeysFromPluginModule(modules.get(name))], []);
+
+        const activePlugins = Array.from(plugins).filter(plugin => result.includes(plugin));
+        const activePluginConfigurationKeys = activePlugins.reduce<Array<string>>((allKeys, name) =>
+          [...allKeys, ...getConfigurationKeysFromPluginModule(modules.get(name))], []);
+
+        const keysToRemove = allPluginConfigurationKeys.filter(key => !activePluginConfigurationKeys.includes(key));
+        await Configuration.removeConfigurationEntries(tmpDir, keysToRemove);
```

Comment:
I'm not sure it's a good idea to remove the settings. For example, let's say I write an IPFS fetcher plugin and want to use `dlx`. In this scenario, I'd likely want to configure the plugin and have it apply to the `dlx` command, right?

I think we should instead just map the plugin paths, doing something like this (note that the `plugin` field accepts both a string and an object with a `path` property):

```ts
configuration.plugins = configuration.plugins.map(plugin => {
  return ppath.resolve(configuration.projectCwd, plugin);
});
```

Reply:
Definitely! I think I implemented it correctly now. Both a direct string and an object with a path entry are supported, and before any rewriting is done it checks whether the file exists (so that direct plugin URLs as described on [this page](https://yarnpkg.com/features/plugins)) continue to work. Is this roughly what you had imagined?

Reply:
Yep, looks much better!

> (so that direct plugin URLs as described on this page continue to work)

Do you mean that?:

> By using yarn plugin import with an URL or a filesystem path, you'll be able to download (and execute, be careful!) code provided by anyone.

If so, `yarn plugin import` will download the URLs and write the path into the file, so it'll never be anything else than a path - no need for the `exists`, since the Configuration loading would have thrown an error if a plugin is invalid 🙂

Reply:
Yeah, that's what I meant! Good to know, that makes much more sense. I removed it now. Anything else left you guys want to have changed?

By the way, thank you so much for leading me through this! I feel like I'm well-prepared to tackle other issues now 😄